google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/apis/GoogleCloudApiSelectorPanel.java

```diff
@@ -184,17 +184,29 @@ private void createUIComponents() {
                 int selectedIndex = model.getMinSelectionIndex();
                 CloudLibrary library =
                     (CloudLibrary) cloudLibrariesTable.getModel().getValueAt(selectedIndex, 0);
+                boolean isSelected =
+                    (boolean) cloudLibrariesTable.getModel().getValueAt(selectedIndex, 1);
                 detailsPanel.setCloudLibrary(library, apiManagementMap.get(library));
+                detailsPanel.setManagementUIEnabled(isSelected);
               }
             });
+    cloudLibrariesTable
+        .getModel()
+        .addTableModelListener(
+            e -> {
+              int row = e.getFirstRow();
+              int column = e.getColumn();
+              TableModel model = (TableModel) e.getSource();
+              boolean isSelected = (boolean) model.getValueAt(row, column);
```

Comment:
It feels a bit over complicated here. We have a specific column that we need to check for a boolean value, and we can check for it on currently selected row. Getting first changed row and changed column is very generic and it feels the code might check for many more combinations, plus it explicitly casts any value from table to boolean. How about just getValueAt(getSelectedRow(), [apiSelectedColumn])?

Also, maybe we can use less generic name for `isSelected`? It feels very UI specific.

Reply:
updated